Support parameterized builds with the Rebuild button

### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  <!-- Paste your environment details below -->
  ```text
Jenkins: 2.426.1
OS: Linux - 5.8.7-1.el7.elrepo.x86_64
Java: 11.0.21 - Eclipse Adoptium (OpenJDK 64-Bit Server VM)
---
ace-editor:1.1
active-directory:2.34
analysis-model-api:11.13.0
ansicolor:1.0.4
ant:497.v94e7d9fffa_b_9
antisamy-markup-formatter:162.v0e6ec0fcfcf6
apache-httpcomponents-client-4-api:4.5.14-208.v438351942757
async-http-client:1.9.40.0
authentication-tokens:1.53.v1c90fd9191a_b_
authorize-project:1.7.1
badge:1.9.1
blueocean:1.27.9
blueocean-autofavorite:1.2.5
blueocean-bitbucket-pipeline:1.27.9
blueocean-commons:1.27.9
blueocean-config:1.27.9
blueocean-core-js:1.27.9
blueocean-dashboard:1.27.9
blueocean-display-url:2.4.2
blueocean-events:1.27.9
blueocean-git-pipeline:1.27.9
blueocean-github-pipeline:1.27.9
blueocean-i18n:1.27.9
blueocean-jira:1.27.9
blueocean-jwt:1.27.9
blueocean-personalization:1.27.9
blueocean-pipeline-api-impl:1.27.9
blueocean-pipeline-editor:1.27.9
blueocean-pipeline-scm-api:1.27.9
blueocean-rest:1.27.9
blueocean-rest-impl:1.27.9
blueocean-web:1.27.9
bootstrap4-api:4.6.0-6
bootstrap5-api:5.3.2-2
bouncycastle-api:2.30.1.77-225.v26ea_c9455fd9
branch-api:2.1135.v8de8e7899051
build-blocker-plugin:1.7.9
build-environment:1.7
build-failure-analyzer:2.2.1-DRE4.0.1-fifa
build-monitor-plugin:1.13+build.202109302210
build-name-setter:2.4.0
build-timeout:1.31
buildmetadata-plugin:4.13
caffeine-api:3.1.8-133.v17b_1ff2e0599
checks-api:2.0.2
cloudbees-bitbucket-branch-source:856.v04c46c86f911
cloudbees-disk-usage-simple:187.v6378d330d1d4
cloudbees-folder:6.858.v898218f3609d
clover:4.14.1.577.vb_ff5a_f69b_37b_
command-launcher:107.v773860566e2e
commons-lang3-api:3.13.0-62.v7d18e55f51e2
commons-text-api:1.11.0-95.v22a_d30ee5d36
conditional-buildstep:1.4.3
config-file-provider:959.vcff671a_4518b_
configuration-as-code:1763.vb_fe9c1b_83f7b
configuration-as-code-groovy:1.1
copyartifact:722.v0662a_9b_e22a_c
cors:0.4.0
cors-filter:1.1
credentials:1311.vcf0a_900b_37c2
credentials-binding:642.v737c34dea_6c2
custom-artifact-archive-plugin:0.1
cvs:2.19.1
dark-theme:410.v3095e2c26b_17
data-tables-api:1.13.8-2
display-url-api:2.200.vb_9327d658781
docker-commons:439.va_3cb_0a_6a_fb_29
docker-workflow:572.v950f58993843
drecolumns:DRE2.18
durable-task:523.va_a_22cf15d5e0
echarts-api:5.4.3-2
email-ext:2.102
envinject:2.908.v66a_774b_31d93
envinject-api:1.199.v3ce31253ed13
extended-bfa-plugin:1.00.07
extended-pipeline-basic-steps-plugin:1.01
extended-read-permission:53.v6499940139e5
extra-columns:1.26
favorite:2.208.v91d65b_7792a_c
font-awesome-api:6.5.1-1
forensics-api:2.3.0
git:5.2.1
git-client:4.6.0
git-server:99.va_0826a_b_cdfa_d
github:1.37.3.1
github-api:1.318-461.v7a_c09c9fa_d63
github-branch-source:1767.va_7d01ea_c7256
gitlab-plugin:1.7.16
gradle:2.9
greenballs:1.15.1
groovy:457.v99900cb_85593
groovy-label-assignment:1.2.0
groovy-postbuild:228.vcdb_cf7265066
gson-api:2.10.1-3.vb_25b_599b_e4f8
handlebars:3.0.8
handy-uri-templates-2-api:2.1.8-30.v7e777411b_148
hashicorp-vault-plugin:364.vf5d54b_3dc313
htmlpublisher:1.32
http_request:1.18
icon-shim:1.0.3
instance-identity:185.v303dc7c645f9
ionicons-api:56.v1b_1c8c49374e
jackson2-api:2.15.3-372.v309620682326
jacoco:3.3.5
jakarta-activation-api:2.0.1-3
jakarta-mail-api:2.0.1-3
javadoc:243.vb_b_503b_b_45537
javax-activation-api:1.2.0-6
javax-mail-api:1.6.2-9
jaxb:2.3.9-1
jdk-tool:73.vddf737284550
jenkins-design-language:1.27.9
jersey2-api:2.40-1
jira:3.12
jira-steps:2.0.165.v8846cf59f3db
jira-trigger:1.0.3
jjwt-api:0.11.5-77.v646c772fddb_0
job-dsl:1.87
job-summary:1.05
jobConfigHistory:1229.v3039470161a_d
joda-time-api:2.12.5-5.v5495a_235fedf
jquery:1.12.4-1
jquery-detached:1.2.1
jquery3-api:3.7.1-1
jsch:0.2.8-65.v052c39de79b_2
json-api:20231013-3.v20f3c247f2fe
json-path-api:2.8.0-5.v07cb_a_1ca_738c
junit:1252.vfc2e5efa_294f
kubernetes:4147.va_d406fb_66172
kubernetes-client-api:6.8.1-224.vd388fca_4db_3b_
kubernetes-credentials:0.11
ldap:711.vb_d1a_491714dc
lockable-resources:1218.va_3dd45e2b_fa_7
log-parser:2.3.1
login-theme:108.vf74a_83b_2238b_
logstash:2.5.0205.vd05825ed46bd-DRE0.0.3
mailer:463.vedf8358e006b_
mapdb-api:1.0.9-28.vf251ce40855d
mask-passwords:173.v6a_077a_291eb_5
material-theme:0.5.2-rc100.6121925fe229
matrix-auth:3.2.1
matrix-project:822.v01b_8c85d16d2
maven-plugin:3.23
mercurial:1260.vdfb_723cdcc81
metrics:4.2.18-442.v02e107157925
mina-sshd-api-common:2.11.0-86.v836f585d47fa_
mina-sshd-api-core:2.11.0-86.v836f585d47fa_
momentjs:1.1.1
monitoring:1.95.0
multi-slave-config-plugin:1.2.0
multiple-choice-plugin:1.12
naginator:1.436.vb_e769dcb_cdf6
next-executions:239.v27031b_e26418
node-iterator-api:1.5
nodejs:1.6.1
okhttp-api:4.11.0-157.v6852a_a_fa_ec11
opentelemetry:2.19.0
optional-run-parameter-plugin:1.06
p4:1.14.4
p4-script-plugin:1.05.0
pam-auth:1.10
parameterized-trigger:787.v665fcf2a_830b_
pipeline-build-step:540.vb_e8849e1a_b_d8
pipeline-graph-analysis:202.va_d268e64deb_3
pipeline-graph-view:205.vb_8e3a_b_51f12e
pipeline-groovy-lib:689.veec561a_dee13
pipeline-input-step:477.v339683a_8d55e
pipeline-milestone-step:111.v449306f708b_7
pipeline-model-api:2.2151.ve32c9d209a_3f
pipeline-model-declarative-agent:1.0
pipeline-model-definition:2.2151.ve32c9d209a_3f
pipeline-model-extensions:2.2151.ve32c9d209a_3f
pipeline-rest-api:2.34
pipeline-stage-step:305.ve96d0205c1c6
pipeline-stage-tags-metadata:2.2151.ve32c9d209a_3f
pipeline-stage-view:2.34
pipeline-utility-steps:2.16.0
plain-credentials:143.v1b_df8b_d3b_e48
plot:2.1.12
plugin-util-api:3.6.0
popper-api:1.16.1-3
popper2-api:2.11.5-1
powershell:2.1
prism-api:1.29.0-8
prometheus:2.4.1
promoted-builds:940.va_9b_59a_717a_b_1
pubsub-light:1.18
rebuild:330.v645b_7df10e2a_
resource-manager-plugin:5.1.0
role-strategy:689.v731678c3e0eb_
run-condition:1.7
scm-api:683.vb_16722fb_b_80b_
script-security:1294.v99333c047434
sectioned-view:1.25-DRE0.0.1
simple-theme-plugin:176.v39740c03a_a_f5
slack:684.v833089650554
snakeyaml-api:2.2-111.vc6598e30cc65
solarized-theme:0.1
sse-gateway:1.26
ssh-credentials:308.ve4497b_ccd8f4
ssh-slaves:2.947.v64ee6b_f87b_c1
sshd:3.242.va_db_9da_b_26a_c3
status-buildname-plugin:1.01
strict-crumb-issuer:2.1.1
structs:325.vcb_307d2a_2782
subversion:2.17.3
support-core:1366.v9d076592655d
swarm:3.333-SNAPSHOT (private-a84f7ce1-root)
synopsys-coverity:3.0.3
theme-manager:215.vc1ff18d67920
throttle-concurrents:2.14
timestamp-fixer:DRE1.00
timestamper:1.26
token-macro:400.v35420b_922dcb_
translation:1.16
trilead-api:2.133.vfb_8a_7b_9c5dd1
update-sites-manager:2.0.0
variant:60.v7290fc0eb_b_cd
view-job-filters:369.ve0513a_a_f5524
vsphere-cloud:2.27
warnings-ng:10.5.2
windows-slaves:1.8.1
workflow-aggregator:596.v8c21c963d92d
workflow-api:1283.v99c10937efcb_
workflow-basic-steps:1042.ve7b_140c4a_e0c
workflow-cps:3826.v3b_5707fe44da_
workflow-cps-global-lib:2.19
workflow-durable-task-step:1313.vcb_970b_d2a_fb_3
workflow-job:1385.vb_58b_86ea_fff1
workflow-multibranch:756.v891d88f2cd46
workflow-scm-step:415.v434365564324
workflow-step-api:639.v6eca_cd8c04a_a_
workflow-support:865.v43e78cc44e0d
zentimestamp:4.2

  ```

</details>


### What Operating System are you using (both controller, and any agents involved in the problem)?

OS: Linux - 5.8.7-1.el7.elrepo.x86_64

### Reproduction steps

- Build a parameterized job
- Click on the `Rebuild` button from either the Pipeline Graph View or Pipeline Console View
- Note that a new build isn't started

### Expected Results

The user is redirected to the build with parameters page.

### Actual Results

Nothing appears to happen, but there is an exception:

<details>
  <summary>Stack Trace</summary>

  <!-- Paste your environment details below -->
  ```text
java.lang.reflect.InvocationTargetException
    at org.kohsuke.stapler.Function$MethodFunction.invoke (Function.java:401)
    at org.kohsuke.stapler.Function$InstanceFunction.invoke (Function.java:409)
    at org.kohsuke.stapler.Function.bindAndInvoke (Function.java:207)
    at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse (Function.java:140)
    at org.kohsuke.stapler.MetaClass$11.doDispatch (MetaClass.java:558)
    at org.kohsuke.stapler.NameBasedDispatcher.dispatch (NameBasedDispatcher.java:59)
    at org.kohsuke.stapler.Stapler.tryInvoke (Stapler.java:770)
    at org.kohsuke.stapler.Stapler.invoke (Stapler.java:900)
    at org.kohsuke.stapler.MetaClass$4.doDispatch (MetaClass.java:289)
    at org.kohsuke.stapler.NameBasedDispatcher.dispatch (NameBasedDispatcher.java:59)
    at org.kohsuke.stapler.Stapler.tryInvoke (Stapler.java:770)
    at org.kohsuke.stapler.Stapler.invoke (Stapler.java:900)
    at org.kohsuke.stapler.Stapler.invoke (Stapler.java:698)
    at org.kohsuke.stapler.Stapler.service (Stapler.java:248)
    at javax.servlet.http.HttpServlet.service (HttpServlet.java:590)
    at org.eclipse.jetty.servlet.ServletHolder.handle (ServletHolder.java:764)
    at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter (ServletHandler.java:1665)
    at hudson.util.PluginServletFilter$1.doFilter (PluginServletFilter.java:163)
    at jenkins.metrics.impl.MetricsFilter.doFilter (MetricsFilter.java:125)
    at hudson.util.PluginServletFilter$1.doFilter (PluginServletFilter.java:160)
    at jenkins.util.HttpServletFilter$1.doFilter (HttpServletFilter.java:76)
    at hudson.util.PluginServletFilter$1.doFilter (PluginServletFilter.java:160)
    at hudson.util.PluginServletFilter.doFilter (PluginServletFilter.java:166)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at jenkins.ErrorAttributeFilter.doFilter (ErrorAttributeFilter.java:29)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at hudson.security.csrf.CrumbFilter.doFilter (CrumbFilter.java:154)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at hudson.security.ChainedServletFilter$1.doFilter (ChainedServletFilter.java:94)
    at hudson.security.ChainedServletFilter.doFilter (ChainedServletFilter.java:111)
    at hudson.security.HudsonFilter.doFilter (HudsonFilter.java:172)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at org.kohsuke.stapler.compression.CompressionFilter.doFilter (CompressionFilter.java:53)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at hudson.util.CharacterEncodingFilter.doFilter (CharacterEncodingFilter.java:86)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter (DiagnosticThreadNameFilter.java:30)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at jenkins.security.SuspiciousRequestFilter.doFilter (SuspiciousRequestFilter.java:38)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle (ServletHandler.java:527)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:131)
    at org.eclipse.jetty.security.SecurityHandler.handle (SecurityHandler.java:549)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:122)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:223)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle (SessionHandler.java:1570)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:221)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle (ContextHandler.java:1384)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:176)
    at org.eclipse.jetty.servlet.ServletHandler.doScope (ServletHandler.java:484)
    at org.eclipse.jetty.server.session.SessionHandler.doScope (SessionHandler.java:1543)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:174)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope (ContextHandler.java:1306)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:129)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle (ContextHandlerCollection.java:149)
    at org.eclipse.jetty.server.handler.HandlerList.handle (HandlerList.java:51)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:122)
    at org.eclipse.jetty.server.Server.handle (Server.java:563)
    at org.eclipse.jetty.server.HttpChannel.lambda$handle$0 (HttpChannel.java:505)
    at org.eclipse.jetty.server.HttpChannel.dispatch (HttpChannel.java:762)
    at org.eclipse.jetty.server.HttpChannel.handle (HttpChannel.java:497)
    at org.eclipse.jetty.server.HttpConnection.onFillable (HttpConnection.java:282)
    at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded (AbstractConnection.java:314)
    at org.eclipse.jetty.io.FillInterest.fillable (FillInterest.java:100)
    at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run (SelectableChannelEndPoint.java:53)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask (AdaptiveExecutionStrategy.java:416)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask (AdaptiveExecutionStrategy.java:385)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce (AdaptiveExecutionStrategy.java:272)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0 (AdaptiveExecutionStrategy.java:140)
    at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run (ReservedThreadExecutor.java:411)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob (QueuedThreadPool.java:969)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob (QueuedThreadPool.java:1194)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run (QueuedThreadPool.java:1149)
    at java.lang.Thread.run (Thread.java:829)
Caused by: javax.servlet.ServletException: This page expects a form submission but had only {delay=[0sec]}
    at org.kohsuke.stapler.RequestImpl.getSubmittedForm (RequestImpl.java:1137)
    at hudson.model.ParametersDefinitionProperty._doBuild (ParametersDefinitionProperty.java:155)
    at jenkins.model.ParameterizedJobMixIn.doBuild (ParameterizedJobMixIn.java:214)
    at jenkins.model.ParameterizedJobMixIn$ParameterizedJob.doBuild (ParameterizedJobMixIn.java:404)
    at java.lang.invoke.MethodHandle.invokeWithArguments (MethodHandle.java:710)
    at org.kohsuke.stapler.Function$MethodFunction.invoke (Function.java:397)
    at org.kohsuke.stapler.Function$InstanceFunction.invoke (Function.java:409)
    at org.kohsuke.stapler.Function.bindAndInvoke (Function.java:207)
    at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse (Function.java:140)
    at org.kohsuke.stapler.MetaClass$11.doDispatch (MetaClass.java:558)
    at org.kohsuke.stapler.NameBasedDispatcher.dispatch (NameBasedDispatcher.java:59)
    at org.kohsuke.stapler.Stapler.tryInvoke (Stapler.java:770)
    at org.kohsuke.stapler.Stapler.invoke (Stapler.java:900)
    at org.kohsuke.stapler.MetaClass$4.doDispatch (MetaClass.java:289)
    at org.kohsuke.stapler.NameBasedDispatcher.dispatch (NameBasedDispatcher.java:59)
    at org.kohsuke.stapler.Stapler.tryInvoke (Stapler.java:770)
    at org.kohsuke.stapler.Stapler.invoke (Stapler.java:900)
    at org.kohsuke.stapler.Stapler.invoke (Stapler.java:698)
    at org.kohsuke.stapler.Stapler.service (Stapler.java:248)
    at javax.servlet.http.HttpServlet.service (HttpServlet.java:590)
    at org.eclipse.jetty.servlet.ServletHolder.handle (ServletHolder.java:764)
    at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter (ServletHandler.java:1665)
    at hudson.util.PluginServletFilter$1.doFilter (PluginServletFilter.java:163)
    at jenkins.metrics.impl.MetricsFilter.doFilter (MetricsFilter.java:125)
    at hudson.util.PluginServletFilter$1.doFilter (PluginServletFilter.java:160)
    at jenkins.util.HttpServletFilter$1.doFilter (HttpServletFilter.java:76)
    at hudson.util.PluginServletFilter$1.doFilter (PluginServletFilter.java:160)
    at hudson.util.PluginServletFilter.doFilter (PluginServletFilter.java:166)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at jenkins.ErrorAttributeFilter.doFilter (ErrorAttributeFilter.java:29)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at hudson.security.csrf.CrumbFilter.doFilter (CrumbFilter.java:154)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at hudson.security.ChainedServletFilter$1.doFilter (ChainedServletFilter.java:94)
    at hudson.security.ChainedServletFilter.doFilter (ChainedServletFilter.java:111)
    at hudson.security.HudsonFilter.doFilter (HudsonFilter.java:172)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at org.kohsuke.stapler.compression.CompressionFilter.doFilter (CompressionFilter.java:53)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at hudson.util.CharacterEncodingFilter.doFilter (CharacterEncodingFilter.java:86)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter (DiagnosticThreadNameFilter.java:30)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at jenkins.security.SuspiciousRequestFilter.doFilter (SuspiciousRequestFilter.java:38)
    at org.eclipse.jetty.servlet.FilterHolder.doFilter (FilterHolder.java:202)
    at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter (ServletHandler.java:1635)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle (ServletHandler.java:527)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:131)
    at org.eclipse.jetty.security.SecurityHandler.handle (SecurityHandler.java:549)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:122)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:223)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle (SessionHandler.java:1570)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle (ScopedHandler.java:221)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle (ContextHandler.java:1384)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:176)
    at org.eclipse.jetty.servlet.ServletHandler.doScope (ServletHandler.java:484)
    at org.eclipse.jetty.server.session.SessionHandler.doScope (SessionHandler.java:1543)
    at org.eclipse.jetty.server.handler.ScopedHandler.nextScope (ScopedHandler.java:174)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope (ContextHandler.java:1306)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle (ScopedHandler.java:129)
    at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle (ContextHandlerCollection.java:149)
    at org.eclipse.jetty.server.handler.HandlerList.handle (HandlerList.java:51)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle (HandlerWrapper.java:122)
    at org.eclipse.jetty.server.Server.handle (Server.java:563)
    at org.eclipse.jetty.server.HttpChannel.lambda$handle$0 (HttpChannel.java:505)
    at org.eclipse.jetty.server.HttpChannel.dispatch (HttpChannel.java:762)
    at org.eclipse.jetty.server.HttpChannel.handle (HttpChannel.java:497)
    at org.eclipse.jetty.server.HttpConnection.onFillable (HttpConnection.java:282)
    at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded (AbstractConnection.java:314)
    at org.eclipse.jetty.io.FillInterest.fillable (FillInterest.java:100)
    at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run (SelectableChannelEndPoint.java:53)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask (AdaptiveExecutionStrategy.java:416)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask (AdaptiveExecutionStrategy.java:385)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce (AdaptiveExecutionStrategy.java:272)
    at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0 (AdaptiveExecutionStrategy.java:140)
    at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run (ReservedThreadExecutor.java:411)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob (QueuedThreadPool.java:969)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob (QueuedThreadPool.java:1194)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run (QueuedThreadPool.java:1149)
    at java.lang.Thread.run (Thread.java:829)
    ```
</details>


### Anything else?

The `Rebuild` button currently redirects to `../../build`, which should be 

### Are you interested in contributing a fix?

_No response_

--- a/src/main/java/io/jenkins/plugins/pipelinegraphview/utils/AbstractPipelineViewAction.java
+++ b/src/main/java/io/jenkins/plugins/pipelinegraphview/utils/AbstractPipelineViewAction.java
@@ -4,6 +4,7 @@ import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import hudson.model.Action;
 import hudson.model.BallColor;
+import hudson.model.ParametersDefinitionProperty;
 import hudson.security.Permission;
 import hudson.util.HttpResponses;
 import java.util.concurrent.ExecutionException;
@@ -43,6 +44,11 @@ public abstract class AbstractPipelineViewAction implements Action, IconSpec {
 
     public String getBuildDisplayName() {
         return run.getDisplayName();
+    }
+
+    public boolean isParameterized() {
+        ParametersDefinitionProperty property = run.getParent().getProperty(ParametersDefinitionProperty.class);
+        return property != null && !property.getParameterDefinitions().isEmpty();
     }
 
     public String getFullBuildDisplayName() {

--- a/src/main/resources/io/jenkins/plugins/pipelinegraphview/PipelineGraphViewAction/index.jelly
+++ b/src/main/resources/io/jenkins/plugins/pipelinegraphview/PipelineGraphViewAction/index.jelly
@@ -27,7 +27,7 @@
           <j:if test="${it.buildable}">
             <l:hasPermission permission="${it.permission}">
               <button id="pgv-rebuild" data-success-message="${%Build scheduled}" data-build-path="../../build"
-                      class="jenkins-button jenkins-!-build-color">
+                      data-parameterized="${it.parameterized}" class="jenkins-button jenkins-!-build-color">
                 <l:icon src="symbol-play-outline plugin-ionicons-api"/>
                 ${%Rebuild}
               </button>

--- a/src/main/resources/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction/index.jelly
+++ b/src/main/resources/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction/index.jelly
@@ -25,7 +25,8 @@
                     <j:if test="${it.buildable}">
                         <l:hasPermission permission="${it.permission}">
                             <button id="pgv-rebuild" data-success-message="${%Build scheduled}"
-                                    data-build-path="../../build" class="jenkins-button jenkins-!-build-color">
+                                    data-build-path="../../build" data-parameterized="${it.parameterized}"
+                                    class="jenkins-button jenkins-!-build-color">
                                 <l:icon src="symbol-play-outline plugin-ionicons-api"/>
                                 ${%Rebuild}
                             </button>

--- a/src/main/resources/io/jenkins/plugins/pipelinegraphview/multipipelinegraphview/MultiPipelineGraphViewAction/index.jelly
+++ b/src/main/resources/io/jenkins/plugins/pipelinegraphview/multipipelinegraphview/MultiPipelineGraphViewAction/index.jelly
@@ -8,7 +8,7 @@
         <j:if test="${it.buildable}">
           <l:hasPermission permission="${it.permission}">
             <button id="pgv-rebuild" data-success-message="${%Build scheduled}" data-build-path="../build"
-                    class="jenkins-button jenkins-!-build-color">
+                    data-parameterized="${it.parameterized}" class="jenkins-button jenkins-!-build-color">
               <l:icon src="symbol-play-outline plugin-ionicons-api"/>
               ${%Build}
             </button>

--- a/src/main/webapp/js/build.js
+++ b/src/main/webapp/js/build.js
@@ -3,18 +3,23 @@ const rebuildButton = document.getElementById('pgv-rebuild');
 if (rebuildButton) {
   rebuildButton.addEventListener('click', event => {
     event.preventDefault();
-    fetch(`${rebuildButton.dataset.buildPath}?delay=0sec`, {
-      method: 'post',
-      headers: {
-        [document.head.dataset.crumbHeader]: document.head.dataset.crumbValue
-      }
-    })
-      .then(res => {
-        if (!res.ok) {
-          console.error('Build failed', res);
-        } else {
-          window.hoverNotification(rebuildButton.dataset.successMessage, rebuildButton);
+    const buildUrl = `${rebuildButton.dataset.buildPath}?delay=0sec`
+    if (rebuildButton.dataset.parameterized === 'true') {
+      window.location.href = buildUrl
+    } else {
+      fetch(buildUrl, {
+        method: 'post',
+        headers: {
+          [document.head.dataset.crumbHeader]: document.head.dataset.crumbValue
         }
       })
+        .then(res => {
+          if (!res.ok) {
+            console.error('Build failed', res);
+          } else {
+            window.hoverNotification(rebuildButton.dataset.successMessage, rebuildButton);
+          }
+        })
+    }
   })
 }

--- a/src/test/java/io/jenkins/plugins/pipelinegraphview/PipelineGraphViewRebuildTest.java
+++ b/src/test/java/io/jenkins/plugins/pipelinegraphview/PipelineGraphViewRebuildTest.java
@@ -1,0 +1,49 @@
+package io.jenkins.plugins.pipelinegraphview;
+
+import static org.junit.Assert.assertEquals;
+
+import hudson.model.Result;
+import io.jenkins.plugins.pipelinegraphview.utils.TestUtils;
+import org.htmlunit.html.HtmlPage;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class PipelineGraphViewRebuildTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Issue("GH#330")
+    @Test
+    public void rebuildButtonStartsNewBuild() throws Exception {
+        WorkflowRun run =
+                TestUtils.createAndRunJob(j, "hello_world", "helloWorldScriptedPipeline.jenkinsfile", Result.SUCCESS);
+
+        try (JenkinsRule.WebClient webClient = j.createWebClient()) {
+            HtmlPage page = webClient.getPage(run, new PipelineGraphViewAction(run).getUrlName());
+            HtmlPage newPage = page.getElementById("pgv-rebuild").click();
+            assertEquals(page.getBaseURL(), newPage.getBaseURL());
+        }
+    }
+
+    @Issue("GH#330")
+    @Test
+    public void rebuildButtonRedirectsForParameterizedJob() throws Exception {
+        WorkflowRun run = TestUtils.createAndRunJob(
+                j, "echo_parameterized", "gh330_parameterizedBuild.jenkinsfile", Result.SUCCESS);
+
+        try (JenkinsRule.WebClient webClient = j.createWebClient()) {
+            // the build page is returned with a 405 Method Not Allowed status code, but the page
+            // exists and can be worked with; we should not fail on this status code
+            webClient.getOptions().setThrowExceptionOnFailingStatusCode(false);
+
+            HtmlPage page = webClient.getPage(run, new PipelineGraphViewAction(run).getUrlName());
+            HtmlPage newPage = page.getElementById("pgv-rebuild").click();
+            assertEquals(
+                    j.getURL() + run.getParent().getUrl() + "build?delay=0sec",
+                    newPage.getBaseURL().toExternalForm());
+        }
+    }
+}

--- a/src/test/resources/io/jenkins/plugins/pipelinegraphview/utils/gh330_parameterizedBuild.jenkinsfile
+++ b/src/test/resources/io/jenkins/plugins/pipelinegraphview/utils/gh330_parameterizedBuild.jenkinsfile
@@ -1,0 +1,13 @@
+pipeline {
+    agent any
+    parameters {
+        string(name: 'MESSAGE', defaultValue: 'Hello world!')
+    }
+    stages {
+        stage('Echo') {
+            steps {
+                echo params.MESSAGE
+            }
+        }
+    }
+}

